src/main/java/org/wso2/securevault/commons/MiscellaneousUtil.java

```diff
@@ -1,329 +1,349 @@
+            xmlSecretResolver = (XMLSecretResolver) secretResolver;
+        } else {
+            throw new SecureVaultException("Secret resolver type mismatch. Require: " + XMLSecretResolver.class + " "
+                    + "found: " + secretResolver.getClass());
+        }
+        OMAttribute attribute = omElement.getAttribute(
+                new QName(xmlSecretResolver.getSecureVaultNamespace(),
+                        xmlSecretResolver.getSecureVaultAlias()));
+        if (attribute != null && attribute.getAttributeValue() != null
+                && !attribute.getAttributeValue().isEmpty()) {
+            if (secretResolver.isTokenProtected(attribute.getAttributeValue())) {
+                resolvedValue = resolve(attribute, xmlSecretResolver);
+            } else {
+                resolvedValue = omElement.getText();
+            }
+        } else {
+            value = omElement.getText();
+            resolvedValue = resolve(value, xmlSecretResolver);
+        }
+        return resolvedValue;
+    }
+
+    public static String resolve(String inputText, SecretResolver secretResolver) {
+
+        String resolvedValue;
+        if (!(secretResolver != null && secretResolver.isInitialized())) {
+            return inputText;
+        }
+        List<ProtectedToken> tokenList = extractProtectedTokens(inputText);
+        if (tokenList.isEmpty()) {
+            if (secretResolver.isTokenProtected(inputText)) {
+                return secretResolver.resolve(inputText);
+            }
+            return inputText;
+        }
+
+        StringBuilder resolvedValueBuilder = new StringBuilder(inputText);
+        for (int i = tokenList.size() - 1; i > -1; i--) {
+            ProtectedToken token = tokenList.get(i);
+            if (secretResolver.isTokenProtected(token.getValue())) {
+                String decryptedValue = secretResolver.resolve(token.getValue());
+                resolvedValueBuilder.replace(token.getStartIndex(), token.getEndIndex() + 1, decryptedValue);
+            }
+        }
+        resolvedValue = resolvedValueBuilder.toString();
+        return resolvedValue;
+    }
+
+    public static String resolve(OMAttribute attribute, SecretResolver secretResolver) {
+
+        String value = attribute.getAttributeValue();
+        return resolve(value, secretResolver);
+    }
+
+    /**
+     * Validate the property value to avoid the processing of null values.
+     *
+     * @param propValue Value of the required property.
+     * @return Return true if not null.
+     */
+    public static boolean isValidPropertyValue(String propValue) {
+
+        if (propValue == null || "".equals(propValue)) {
+            if (log.isDebugEnabled()) {
+                log.debug("No secret repository has been configured.");
+            }
+            return false;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Configured secret repositories: " + propValue);
```

Comment:
debug log should be changed considering the prop value

Reply:
Modified